scripts/fetchai_code_quality/internal/static_analysis.py

```diff
@@ -127,35 +235,121 @@ def get_disabled_checks_list():
     return sorted(disabled_checks)
 
 
-def static_analysis(project_root, build_root, fix, concurrency):
-    output_file = abspath(join(build_root, 'clang_tidy_fixes.yaml'))
-    runner_script_path = shutil.which('run-clang-tidy-6.0.py')
-    assert runner_script_path is not None
-    clang_tidy_path = shutil.which('clang-tidy-6.0')
-    clang_apply_replacements_path = shutil.which(
-        'clang-apply-replacements-6.0')
-    if fix:
-        assert clang_apply_replacements_path is not None, \
-            'clang-apply-replacements-6.0 must be installed and ' \
-            'found in the path (install clang-tools-6.0)'
+def get_changed_paths_from_git(project_root, commit):
+    raw_relative_paths = subprocess.check_output(['git', 'diff', '--name-only', commit]) \
+        .strip() \
+        .decode('utf-8') \
+        .split('\n')
+
+    relative_paths = [rel_path.strip() for rel_path in raw_relative_paths]
+
+    return [abspath(join(project_root, rel_path)) for rel_path in relative_paths]
+
+
+def filter_non_matching(file_list, words):
+    return [x for x in file_list if any(file_ending in x for file_ending in words)]
+
+
+def filter_compile_commands(project_root, input_location, target_files, output_location):
+    filtered_commands = []
+
+    print(target_files)
+
+    # read the input file and filter the entries
+    with open(input_location, 'r') as input_file:
+        commands = json.load(input_file)
+
+        for command in commands:
+            command_file_path = command['file']
+            relative_file_path = os.path.relpath(
+                command_file_path, project_root)
+
+            if relative_file_path.startswith('vendor'):
+                print('Dropping 2', relative_file_path)
+                continue
+
+            if relative_file_path == 'libs/vm/src/tokeniser.cpp':
+                continue
+
+            if target_files is not None:
+                match = False
+                for target_file in target_files:
+                    if target_file in command_file_path:
+                        match = True
+                        break
+
+                if not match:
+                    print('Dropping 1', relative_file_path)
```

Comment:
dropping 2 and dropping 1 intended to be in?

Reply:
Nope good spot